modules/components/src/DataTable/TableToolbar/TableToolbar.js

```diff
@@ -123,175 +124,176 @@ const TableToolbar = ({
-              itemSelectionLegend={`Select columns to display`}
-              selectAllAriaLabel={`Select all columns`}
-              resetToDefaultAriaLabel={`Reset to default columns`}
-              itemToString={(i) => i.displayName}
-              items={canChangeShowColumns}
-              defaultColumns={defaultColumns}
-              onChange={(item) => {
-                setFilterVal('');
-                onFilterChange({
-                  value: '',
-                  generateNextSQON: generateNextSQON(''),
-                });
-                onColumnsChange({ ...item, show: !item.show });
-              }}
-              onMultipleChange={(changes) => {
-                onMultipleColumnsChange(changes);
-              }}
-              theme={themeDropDownProps}
-            >
-              {columnDropdownText}
-            </MultiSelectDropDown>
-          ) : (
-            <DropDown
-              aria-label={`Select columns`}
-              itemToString={(i) => i.displayName}
-              items={canChangeShowColumns}
-              onChange={(item) => {
-                setFilterVal('');
-                onFilterChange({
-                  value: '',
-                  generateNextSQON: generateNextSQON(''),
-                });
-                onColumnsChange({ ...item, show: !item.show });
-              }}
-              theme={themeDropDownProps}
-            >
-              {columnDropdownText}
-            </DropDown>
-          ))}
+			<div className="group">
+				{allowTogglingColumns &&
+					(enableDropDownControls ? (
+						<MultiSelectDropDown
+							allowControls
+							allowSelection
+							buttonAriaLabelClosed={`Open column selection menu`}
+							buttonAriaLabelOpen={`Close column selection menu`}
+							itemSelectionLegend={`Select columns to display`}
+							selectAllAriaLabel={`Select all columns`}
+							resetToDefaultAriaLabel={`Reset to default columns`}
+							itemToString={(i) => i.displayName}
+							items={canChangeShowColumns}
+							defaultColumns={defaultColumns}
+							onChange={(item) => {
+								setFilterVal('');
+								onFilterChange({
+									value: '',
+									generateNextSQON: generateNextSQON(''),
+								});
+								onColumnsChange({ ...item, show: !item.show });
+							}}
+							onMultipleChange={(changes) => {
+								onMultipleColumnsChange(changes);
+							}}
+							theme={themeDropDownProps}
+						>
+							{columnDropdownText}
+						</MultiSelectDropDown>
+					) : (
+						<DropDown
+							aria-label={`Select columns`}
+							itemToString={(i) => i.displayName}
+							items={canChangeShowColumns}
+							onChange={(item) => {
+								setFilterVal('');
+								onFilterChange({
+									value: '',
+									generateNextSQON: generateNextSQON(''),
+								});
+								onColumnsChange({ ...item, show: !item.show });
+							}}
+							theme={themeDropDownProps}
+							enableSelectColumnDropdownTextFilter={enableSelectColumnDropdownTextFilter}
```

Comment:
```suggestion
							aria-label={`Select columns`}
							enableSelectColumnDropdownTextFilter={enableSelectColumnDropdownTextFilter}
							itemToString={(i) => i.displayName}
							items={canChangeShowColumns}
							onChange={(item) => {
								setFilterVal('');
								onFilterChange({
									value: '',
									generateNextSQON: generateNextSQON(''),
								});
								onColumnsChange({ ...item, show: !item.show });
							}}
							theme={themeDropDownProps}
```

To keep things consistent, we aim to have props alphabetised 💪

Reply:
code changes as suggested

Reply:
awesome!
lol but remind me to show you how to accept commit suggestions (mine there was mergeable into your pr)